File: InfoClientProxy.cs
using CevaTema.Commons;
using CevaTema.Entries;
using CevaTema.MessageMarshaller;
using CevaTema.RequestReply;

namespace CevaTema {
	public class InfoClientProxy : IInfo, IClientProxy {
		private readonly Requestor _req = new Requestor("InfoClientProxy");
		private readonly int _portNumber;

		public InfoClientProxy(int portNumber) {
			_portNumber = portNumber;
		}
		public System.Int32 get_temp(System.String param0) {
			var msgData = param0+":get_temp";
			//scrie in mesaj toti parametri
			var msg = new Message("InfoClientProxy",msgData);
			var bytes = Marshaller.Marshal(msg);
			IAddress dest = new Entry("127.0.0.1",_portNumber);
			//asteapta rezutatul
			bytes = _req.deliver_and_wait_feedback(dest,bytes);
			//despacheteaza rezultatul
			var answer = Marshaller.Unmarshal(bytes);
			//rezultatul este convertiti la tipul care este returnat de metoda curenta
			var result = int.Parse(answer.Data);
			return result;
		}
		public System.String get_road_info(System.Int32 param0) {
			var msgData = param0+":get_road_info";
			//scrie in mesaj toti parametri
			var msg = new Message("InfoClientProxy",msgData);
			var bytes = Marshaller.Marshal(msg);
			IAddress dest = new Entry("127.0.0.1",_portNumber);
			//asteapta rezutatul
			bytes = _req.deliver_and_wait_feedback(dest,bytes);
			//despacheteaza rezultatul
			var answer = Marshaller.Unmarshal(bytes);
			//rezultatul este convertiti la tipul care este returnat de metoda curenta
			var result = answer.Data;
			return result;
		}
	}
}

File: InfoClientProxy.java
import MessageMarshaller.*;
import Commons.Address;
import Registry.Entry;
import RequestReply.*;

class InfoClientProxy implements Info,ClientProxy {
	private Requestor req = new Requestor("InfoClientProxy");
	private Marshaller m = new Marshaller();
	private int portNumber;

	public InfoClientProxy(int portNumber) {
		this.portNumber = portNumber;
	}

	public int get_temp(java.lang.String $param_0) {
		String msgData = $param_0+":get_temp";
		//scrie in mesaj toti parametri
		Message msg = new Message("InfoClientProxy",msgData);
		byte[] bytes = m.marshal(msg);
		Address dest = new Entry("127.0.0.1",portNumber);
		//asteapta rezutatul
		bytes = req.deliver_and_wait_feedback(dest,bytes);
		//despacheteaza rezultatul
		Message answer = m.unmarshal(bytes);
		//rezultatul este convertiti la tipul care este returnat de metoda curenta
		int $result = Integer.parseInt(answer.data);
		return $result;
	}

	public java.lang.String get_road_info(int $param_0) {
		String msgData = String.valueOf($param_0)+":get_road_info";
		//scrie in mesaj toti parametri
		Message msg = new Message("InfoClientProxy",msgData);
		byte[] bytes = m.marshal(msg);
		Address dest = new Entry("127.0.0.1",portNumber);
		//asteapta rezutatul
		bytes = req.deliver_and_wait_feedback(dest,bytes);
		//despacheteaza rezultatul
		Message answer = m.unmarshal(bytes);
		//rezultatul este convertiti la tipul care este returnat de metoda curenta
		java.lang.String $result = answer.data;
		return $result;
	}

}
